src/site_upload/unzip_upload/unzip_upload.py

```diff
@@ -0,0 +1,60 @@
+"""Lambda for moving data to processing locations"""
+
+import datetime
+import logging
+import os
+import zipfile
+from io import BytesIO
+
+import boto3
+
+from shared import decorators, enums, functions
+
+log_level = os.environ.get("LAMBDA_LOG_LEVEL", "INFO")
+logger = logging.getLogger()
+logger.setLevel(log_level)
+
+
+def unzip_upload(s3_client, sns_client, s3_bucket_name: str, s3_key: str) -> None:
+    metadata = functions.parse_s3_key(s3_key)
+    buffer = BytesIO(s3_client.get_object(Bucket=s3_bucket_name, Key=s3_key)["Body"].read())
+    archive = zipfile.ZipFile(buffer)
+    files = archive.namelist()
+    files.remove("manifest.toml")
+    meta_files = [x for x in files if "__meta_" in x]
+    files = list(set(files) - set(meta_files) - set(["manifest.toml"]))
+    # The manifest will be used as a signal that the extract has finished,
+    # so we'll extract it last in all cases.
+    # TODO: decide on extract location for manifests
+    for file_list in [meta_files, files, ["manifest.toml"]]:
+        for file in file_list:
+            target_folder = (
+                functions.get_folder_from_s3_path(s3_key)
+                .replace(f"{enums.BucketPath.STAGING.value}/", f"{enums.BucketPath.UPLOAD.value}/")
+                .replace(f"/{metadata.study}/", f"/{metadata.study}/{file.split('.')[0]}/")
+            )
+            s3_client.upload_fileobj(
+                archive.open(file), Bucket=s3_bucket_name, Key=f"{target_folder}/{file}"
+            )
+    archive_key = s3_key.replace(
+        f"{enums.BucketPath.STAGING.value}/", f"{enums.BucketPath.ARCHIVE.value}/"
+    )
+    functions.move_s3_file(
+        s3_client=s3_client,
+        s3_bucket_name=s3_bucket_name,
+        old_key=s3_key,
+        new_key=f"{archive_key}.{datetime.datetime.now(datetime.UTC)}",
```

Comment:
nit: maybe call isoformat()? I'm used to the version without the spaces, but I guess avoiding spaces is big 90's energy.